singularity-eos/base/root-finding-1d/root_finding.hpp

```diff
@@ -70,45 +70,45 @@ enum class Status { SUCCESS = 0, FAIL = 1 };
 */
 class RootCounts {
  private:
-  static constexpr int nbins_{15};
+  static constexpr std::size_t nbins_{15};
```

Comment:
Clang didn't care, but gcc did. Changed a bunch of `int`s to `size_t`s to avoid warnings from comparing signed vs unsigned.

Reply:
most of these seem like small-ish numbers, could we get away with `uint32_t`?

Reply:
We probably could. Is there a reason to do so though? This machinery is only turned on for debugging.

Reply:
This was in regard to all of the size_t changes. Are the changes only for debug portions of code?

Reply:
I'd have to go through carefully. I think several are loop variables and could almost certainly be `uint32_t`. But will changing that have any real impact on memory footprint or performance?

Reply:
OK I think making sure `uint32_t` is safe everywhere we want to use it is a bit of a rat's nest. I agree it's worth doing though. I created an issue, but I'm not changing it for now. Better done in a later MR. #441